lib/dpl/provider/chef_supermarket.rb

```diff
@@ -20,8 +21,16 @@ def check_auth
         error "#{options[:client_key]} does not exist" unless ::File.exist?(options[:client_key])
       end
 
+      def cookbook_metadata
+        # credit: https://coderwall.com/p/wwt0sw/easiest-way-to-parse-a-cookbook-s-metadata-file
+        metadata_file = 'metadata.rb'
+        # read in metadata
+        @metadata = Chef::Cookbook::Metadata.new
+        @metadata.from_file(metadata_file)
+      end
+
       def check_app
-        @cookbook_name = options[:cookbook_name] || options[:app]
+        @cookbook_name = options[:cookbook_name] || options[:app] || @metadata.name
```

Comment:
The reason that it doesn't work is that this instance variable `@metadata` is never initialized when the execution gets here.